No longer need to include rubygems or bundler software dep

tyler-ball opened the pull request:
## Description
Recent versions of Ruby have included Rubygems and Bundler, which they
did not always do. We also recently fixed some issues in Appbundler so
these software dependencies are no longer needed.

## Related Issue
<!--- If you are suggesting a new feature or change, please create an issue first -->
<!--- Please link to the issue, discourse, or stackoverflow here: -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the **CONTRIBUTING** document.

tyler-ball committed:
No longer need to include rubygems or bundler software dep

Recent versions of Ruby have included Rubygems and Bundler, which they
did not always do. We also recently fixed some issues in Appbundler so
these software dependencies are no longer needed.

Signed-off-by: tyler-ball <tball@chef.io>

james-stocks commented on omnibus_overrides.rb:
```diff
@@ -4,8 +4,6 @@
 #
 # NOTE: You MUST update omnibus-software when adding new versions of
 # software here: bundle exec rake dependencies:update_omnibus_gemfile_lock
```
It may be helpful to insert a blank line to keep the comment about train separate from the comment block above

btm replied:
I added a commit to this PR removing that comment altogether. It's actually not relevant in this project.

btm committed:
Remove comment about the omnibus lock in overrides file

The comment about running `bundle exec rake dependencies:update_omnibus_gemfile_lock`
is copy-pasta from chef/chef. The inspec project does not commit any
Gemfile.lock files, including omnibus/Gemfile.lock, due to
`Gemfile.lock` being in the `.gitignore` file.

Signed-off-by: Bryan McLellan <btm@loftninjas.org>

tas50 commented on omnibus_overrides.rb:
```diff
@@ -1,11 +1,6 @@
 # THIS IS NOW HAND MANAGED, JUST EDIT THE THING
-# .travis.yml and appveyor.yml consume this,
-# try to keep it machine-parsable.
+# .travis.yml and appveyor.yml consume this, try to keep it machine-parsable.
```
This comment is no longer valid. We used it previously to install bundler / rubygems

lamont-granquist replied:
technically its not a bad thing to keep it machine-parseable though.

tas50 replied:
I meant just the bit about travis / appveyor

btm replied:
I removed the whole line. I was going to say "Please try to keep it machine-parsable" but then the obvious question was "for what? how do I know if the machine can parse this or not" and it isn't really doing anyone favors.
